src/assignment/helpers/insert-at-position.ts

```diff
@@ -0,0 +1,33 @@
+type Positioned = any & { id: string; position: number };
+
+export const insertAssignmentBlockAtPosition = (
+	list: Positioned[],
+	item: Positioned,
+	pos: number
+) => {
+	// Pass in the position of the preceding item
+	const before = list.find((x) => x.position === pos);
+
+	// Determine target position
+	const target = before.position + 1;
+
+	// Increment all the following and the matching items
+	const after = list
+		.filter((x) => x.position >= target)
+		.map((x) => ({
+			...x,
+			position: x.position + 1,
+		}));
+
+	return [
+		// Do not touch preceding items
+		...list.filter((x) => x.position < target),
+		// Insert the new item at the position
+		{
+			...item,
+			position: target,
+		},
+		// Add in the following
+		...after,
+	];
```

Comment:
There is already a handy helper for content page blocks that you might be able to generalize:
```typescript
// Helpers
const repositionConfigs = (updatedConfigs: ContentBlockConfig[]) => {
	updatedConfigs.forEach((config, position) => {
		config.position = position;
	});
};
```
to something like:
```typescript
// Helpers
const repositionBlocks= (blocks: {position: number}[]) => {
	blocks.forEach((block, position) => {
		block.position = position;
	});
};
```

Then you can simplify this implementation:
```suggestion
	const newList = [...list];
	newList.splice(pos, 0, item);
	return repositionBlocks(newList);
```

Reply:
🛑 Dit is insert-logica, niet swap-position logica

Reply:
Fixed in https://github.com/viaacode/avo2-client/pull/1367/commits/92d08f043dde87c80ab329c6edb2191311fa7205#diff-d151a59bccd3fce63bd7766e5d1016444fd676c39552421f7f10dff04e7fdc46